apps/src/templates/admin/MassDeleteContainer.tsx

```diff
@@ -0,0 +1,450 @@
+import {Button} from '@code-dot-org/component-library/button';
+import React, {useState} from 'react';
+import * as Table from 'reactabular-table';
+
+import styles from './mass_delete.module.scss';
+
+// Type definitions
+interface OriginalData {
+  student_id?: string;
+  student_username?: string;
+  unit_name: string;
+}
+
+interface ProcessedData {
+  student_id: string;
+  unit_name: string;
+}
+
+// Utility functions
+const parseCsv = (file: File): Promise<OriginalData[]> => {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onload = e => {
+      try {
+        const text = e.target?.result as string;
+        const lines = text.split('\n').filter(line => line.trim());
+
+        if (lines.length < 2) {
+          throw new Error(
+            'CSV must have at least a header row and one data row'
+          );
+        }
+
+        // Parse header row
+        const headers = lines[0]
+          .split(',')
```

Comment:
CSV files can sometimes use other characters as separators (I think I've seen semicolons and possibly tabs). 

I think this works as a hack, however could you test to see if you could use [papaparse](https://www.papaparse.com/), which already looks [installed](https://github.com/code-dot-org/code-dot-org/blob/staging-next/apps/yarn.lock#L8645) and if that doesn't work or gets too hairy just add a comment about the specific CSV format we need.